prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml

```diff
@@ -30,3 +28,25 @@ presubmits:
               - go
               - test
               - ./...
+
+postsubmits:
+    GoogleCloudPlatform/oss-test-infra:
+
+    - name: post-test-infra-deploy-prow
+      cluster: test-infra-trusted
+      run_if_changed: '^prow/cluster/'
+      decorate: true
+      branches:
+      - master
+      labels:
+          preset-prow-deployer-service-account: "true"
+      spec:
+          containers:
+          - image: gcr.io/k8s-testimages/gcloud-in-go:v20190125-cc5d6ecff3
+            command:
+            - make
+            args:
+            - -C
+            - prow
+            - deploy
+            - deploy-build
```

Comment:
Is it safe to use both these targets at the same time? If make builds all the target dependencies first or in parallel then we could unintentionally deploy all the resources to one of the clusters.

Reply:
Make, by *default*, runs each target in a separate subshell ([ref](https://www.gnu.org/software/make/manual/make.html#Execution)). Also, as a additional layer of security in general, we are avoiding updating of *global* state (i.e. `KUBECONFIG`) by using a temp file for each execution.

https://github.com/GoogleCloudPlatform/oss-test-infra/blob/d15f44bc53127d4dc3af73fb0d2abb75f7981d9a/prow/Makefile.gcloud.mk#L24-L27

Reply:
> We should probably move all the "non-trusted" jobs out of the cluster before adding the deployer credentials to it.

In the fine print I mentioned as soon as this merges, I am going to move all "non-trusted" jobs to the `build` cluster (https://github.com/GoogleCloudPlatform/oss-test-infra/pull/99#issue-321850303). Although it may make m ore sense to do it before.

/hold

Reply:
I have migrated "non-trusted" jobs to the `build` cluster and uploaded the `prow-deployer` secret.

/hold cancel